config/routes.rb

```diff
@@ -401,6 +401,15 @@
           put 'test_connection'
         end
         resources :images, :except => [:show]
+        resources :key_pairs, :only => [:index] do
+          member do
+            get 'download'
+            put 'recreate'
+          end
+          collection do
+            delete 'remove'
```

Comment:
why is this still on the collection? you are deleting a specific ID, so the route should be ../key_pairs/ID and the verb is DELETE).